azurerm/data_source_cosmos_db_account.go

```diff
@@ -206,6 +225,16 @@ func dataSourceArmCosmosDBAccountRead(d *schema.ResourceData, meta interface{})
 			return fmt.Errorf("Error setting `capabilities`: %+v", err)
 		}
 
+		virtualNetworkRules := make([]map[string]interface{}, len(*props.VirtualNetworkRules))
```

Comment:
ideally we would break this out into a flatten function.

Also we should be checking if `props.VirtualNetworkRules` is nil before dereferencing it.

Reply:
Done.